lib/sudo-pam/src/rpassword.rs

```diff
@@ -75,13 +75,19 @@ fn safe_tcgetattr(fd: RawFd) -> io::Result<termios> {
 /// Reads a password from the given file descriptor
 fn read_unbuffered(source: &mut impl io::Read) -> io::Result<PamBuffer> {
     let mut password = PamBuffer::default();
+    let mut pwd_iter = password.iter_mut();
 
     const EOL: u8 = 0x0A;
-
-    for (read_byte, dest) in iter::zip(source.bytes(), password.iter_mut()) {
-        match read_byte? {
-            EOL => break,
-            ch => *dest = ch,
+    let input = source.bytes().take_while(|x| x.as_ref().ok() != Some(&EOL));
+
+    for read_byte in input {
+        if let Some(dest) = pwd_iter.next() {
+            *dest = read_byte?
+        } else {
+            return Err(Error::new(
+                ErrorKind::OutOfMemory,
```

Comment:
I'm curious, why `OutOfMemory` ?

Reply:
I have no clear intuitition what to pick here; but since the actual direct cause of having to fail here is the fact that we were trying to read into a buffer that only has space for 511 bytes, and that failed, `OutOfMemory` didn't seem inappropriate. (Note: that specific limit is imposed on us by PAM, but it is also good to have some limit for other reasons)

Reply:
Yeah out-of-memory sounds like a good fit then :)